Extension/browser/firefox/lib/contentPolicy.js

```diff
@@ -517,7 +517,7 @@ var WebRequestImpl = exports.WebRequestImpl = {
 
             var tab = {id: tabUtils.getTabId(xulTab)};
             var requestUrl = newChannel.URI.asciiSpec;
-            var tabUrl = this.framesMap.getFrameUrl(tab, 0);
+            var tabUrl = this.framesMap.getMainFrameUrl(tab);
```

Comment:
Instead of using main frame URL we should use `referrerUrl` of the frame.

The question is how to get to it.

I suppose we should save it to requestProperties and then reuse here (see `this._saveLastRequestProperties`)